runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -444,6 +436,12 @@ public final class MqttSessionProxy
         private int sessionPadding;
         private String willId;
         private int delay;
+        private int unfetchedKafkaTopics;
+        private Long2ObjectHashMap<PublishOffsetMetadata> offsets;
+        private Int2ObjectHashMap<KafkaTopicPartition> partitions;
+        private Int2ObjectHashMap<KafkaTopicPartition> retainedPartitions;
+        private long affinity;
```

Comment:
`affinity` should be `final`, passed to constructor from `newStream`

Reply:
We don't even need to store the affinity in the session stream, as it's only adding the metadata, but never queries. At the beginning. At mqttBegin we already have it.